bin/main.ml

```diff
@@ -13,7 +13,17 @@ let stack_of_addr addr =
   Stack.connect udp_socket tcp_socket
 
 let () =
+  let port =
+    match Sys.getenv_opt "MIRAGE_WWW_PORT" with
+    | Some p -> int_of_string p
+    | None -> 8080
+  in
   Sys.(set_signal sigpipe Signal_ignore);
   Lwt_main.run
   @@ let* stack = stack_of_addr "0.0.0.0/0" in
-     Mirageio.http ~port:8080 stack
+     Lwt.join
```

Comment:
This join is a bit messy and doesn't really help. It looks like `Dream__mirage` doesn't tell you when it's ready to serve files, so you might as well just print the message before starting it.

(ideally, web-servers would take the listening socket as an argument, but for some reason they usually insist on binding the socket themselves)

Reply:
Makes sense.